### ISSUE TITLE
Add 4313 support

### ISSUE THREAD

**Issue by Isaac96** [TCWORLD/ATTinyCore#29]
Please do so :smiley:

**Comment by TCWORLD** [comment]
Add a boards.txt entry using the X313 as the variant, then compile it and see if it works. But the version of avr-gcc that comes with the Arduino IDE doesn't support the 4313 as far as I recall.

**Comment by Isaac96** [comment]
As a matter of fact, none of the boards compile cleanly. I get errors when I try to compile. I pasted them below

Arduino: 1.6.4 (Windows 7), Board: "Optiboot ATtiny84 @ 8 MHz  (internal osc)"

Third-party platform.txt does not define compiler.path. Please report this to the third-party hardware maintainer.

Error while compiling: missing 'recipe.cpp.o.pattern' configuration parameter

**Comment by Isaac96** [comment]
I get the same thing with 1.6.1.

**Comment by TCWORLD** [comment]
Ah, yes. Arduino completely changed the board handling in 1.6.x and I really haven't had the time to look through the required changes to update the core to work with the new arrangement. Fortunately, two people have been able to do the update in their forks: https://github.com/FemtoCow/ATTinyCore/ and https://github.com/SpenceKonde/ATTinyCore/ . For 1.6.x I would recommend using one of these forks until I've had a chance to review the changes and pulled in support. 

Alternatively, it should also be noted that ATTinyCore is integrated into the UECIDE project which is an excellent alternative to the Arduino IDE.

**Comment by ApexMoEng** [comment]
I was trying to search for this issue. Followed threads such as http://forum.arduino.cc/index.php?topic=299759.15

Personally I needed support for ATtiny 167 so I will follow those links you provided.

There definitely needs to be extra .txt files added including .platform.txt.

I tried very hard to get it to work, but I couldn't. Ended up on compiler error.

In file included from C:\Program Files (x86)\Arduino\hardware\tiny\avr\cores\tiny/Arduino.h:222:0,
                 from sketch_jun12a.ino:1:
C:\Program Files (x86)\Arduino\hardware\tiny\avr\cores\tiny/HardwareSerial.h:56:7: error: expected identifier before ')' token
       );
       ^

Edit:
Trying FemtoCow's edits, when i try to verify on some cores it works, when I try to verify on ATtiny 167 though I get compiler errors.

**Comment by samakar** [comment]
I tried you library with Arduino 1.6.5 but generates errors. Is it compatible with 1.6.5?

**Comment by ApexMoEng** [comment]
Try https://github.com/SpenceKonde/ATTinyCore/ He has done the most work on his fork to date.

I worked with him early this month to get his version running on 1.6.0. It hasn't been board tested yet though. Just compiling as far as I know.

**Comment by samakar** [comment]
Hi,
I tested Spence's code but it does not support IDE 1.6.5. Bootloader makes
errors.
Maybe I'd better download the same version you used. What version of IDE
you have tested your code?

**Comment by ApexMoEng** [comment]
1.6.0

And if you are testing his code and have issues you should really start a new thread over on his page. So that he may debug anything and give him feedback.

**Comment by samakar** [comment]
Hi,
 Spence debugged his code and I successfully uploaded a blink sketch to my
Attiny88. Now I want to make SPI sketch. Which spi  library should I use?
Do you have a working code for Attiny88 as a slave?
Thanks.

**Comment by ApexMoEng** [comment]
No. Arduino IDE has a built in SPI library. Again you should open a new thread over on his section.
